flake.nix

```diff
@@ -152,6 +153,89 @@
               ];
               treefmt = pkgs.treefmt;
             };
+            hydra-node = makeTest {
+              name = "hydra-node";
+              nodes = {
+                hydraNode = { ... }: {
+                  environment.systemPackages = [
+                    pkgs.cardano-node
+                    pkgs.cardano-cli
+                    pkgs.check-jsonschema
+                  ];
+                  networking.firewall.enable = false;
+                  imports = [ ];
+                  virtualisation = {
+                    cores = 16;
+                    memorySize = 16384;
+                  };
+                };
+              };
+
+              testScript = ''
+                hydraNode.wait_for_unit("multi-user.target")
+                hydraNode.succeed("cp ${self}/hydra-node/golden . -r")
+                hydraNode.succeed("cp ${self}/hydra-node/json-schemas . -r")
+                hydraNode.succeed("mkdir -p test")
+                hydraNode.succeed("cp ${self}/hydra-node/test/tls test/ -r")
```

Comment:
Could: run the process in the package source directory instead of copying things

I think the `hydra-node` test binary should _just work(tm)_ if you run it in the `hydra-node/` directory.